src/aleph/vm/orchestrator/views/__init__.py

```diff
@@ -318,3 +324,61 @@ async def update_allocations(request: web.Request):
         },
         status=status_code,
     )
+
+
+async def notify_allocation(request: web.Request):
+    if not authenticate_api_request(request):
+        return web.HTTPUnauthorized(text="Authentication token received is invalid")
+
+    try:
+        data = await request.json()
+        vm_notification = VMNotification.parse_obj(data)
+    except ValidationError as error:
+        return web.json_response(data=error.json(), status=web.HTTPBadRequest.status_code)
+
+    pubsub: PubSub = request.app["pubsub"]
+    pool: VmPool = request.app["vm_pool"]
+
+    # First free resources from persistent programs and instances that are not scheduled anymore.
+    instance = vm_notification.instance
```

Comment:
I don't see where you are freeing resources here

Reply:
Here I don't free any resources, this endpoint is only to notify the CRN to create the instance. Anyway, we need to decide first if we will use that endpoint or will use the Scheduler approach.